### PR TITLE
SHRIKE-52: make data integrity validation configurable

### PR DESCRIPTION
**Pull request by dfakhritdinov**

### CODE REVIEW

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala
Comment (mr-git) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala:159]: maybe?
```suggestion
  @deprecated("Use apply1 instead", "3.4.0")
```

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala
Comment (mr-git) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala:203]: why include check for equality? shouldn't it be just `<` instead of `<=`?

File: journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala
Comment (mr-git) [inline journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala:358]: Is `seqNrUniqueness: Boolean` the best we can do? Nicely named ADTs might be more expressive: `CheckSeqNrGrowing` (default) and `IgnoreChecks` (new) 

Frankly, we never check for `Uniqueness` - we check that new `seqNr` larger than previously observed one, but we never cehck for delta to be exactly one, do we? We could add `StrictCheckSeqNr` to do checks for exactly `+1` though...

File: persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournalConfig.scala
Comment (mr-git) [inline persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournalConfig.scala:49]:
```suggestion
  * @param dataIntegrity
```

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala
Comment (dfakhritdinov) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala:203]: Would be better asking author of the code, but AFAIK the reason is in `seqNr` value - its _not_ last seen sequence number but rather next one expected:
```
                  if (seqNr <= record.seqNr) {
                    val seqNr1 = record
                      .seqNr
                      .next[Option]
                    (seqNr1, Stream[F].single(record))
                  }
```
In case of no gaps in sequence numbers would be enough having `seqNr == record.seqNr`, but in reality gaps are happening regularly thus KJ will accept them and delegate accepting/rejecting event to the application.

File: journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala
Comment (dfakhritdinov) [reply journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala:358]: I'm fine with naming and strongly against ADT for true|false values, please do not increase complexity without reason. 

> Frankly, we never check for Uniqueness - we check that new seqNr larger than previously observed one, but we never cehck for delta to be exactly one, do we? We could add StrictCheckSeqNr to do checks for exactly +1 though...

KJ should not check that current `seqNr` is exactly +1 of previous due to possible gaps in the sequence. Now final decision of accepting or rejecting event delegated to the application instead of been forced in KJ.

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala
Comment (mr-git) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala:203]: why gaps happen?

File: persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournalConfig.scala
Comment (mr-git) [inline persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournalConfig.scala:63]: seems that MiMa check wants us to add custom constructor `apply` without `dataIntegrity` - probably we can do that in companion object...

Comment (dfakhritdinov) [reply persistence/src/main/scala/akka/persistence/kafka/journal/KafkaJournalConfig.scala:63]: no, I disabled MiMa for now. Will bring it back after merging the PR

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala
Comment (dfakhritdinov) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/EventualCassandra.scala:203]: Again, the question is not related to the PR. Lets not block it by side discussions.

I suggest asking @theide as one having much more experience in using KJ :D

**Review by mr-git — APPROVED** [review]

**Pull request merged**
